### PR TITLE
[SHELL32] SHChangeNotify: Delete CDirectoryList

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Class `CDirectoryList` is unnecessary and slows down the system.
JIRA issue: [CORE-13950](https://jira.reactos.org/browse/CORE-13950)

## Proposed changes

- Delete `CDirectoryList.cpp` and `CDirectoryList.h`.
- Delete `CDirectoryList` usage.

## TODO

- [x] Do tests.

### CODE REVIEW

**Comment by katahiromz** [comment]
after
Still working.

File: dll/win32/shell32/shelldesktop/CDirectoryWatcher.cpp [hunk 1]
```diff
@@ -171,37 +170,19 @@ void CDirectoryWatcher::ProcessNotification()
         // convert action to event
         fDir = PathIsDirectoryW(szPath);
         dwEvent = ConvertActionToEvent(pInfo->Action, fDir);
-
-        // convert SHCNE_DELETE to SHCNE_RMDIR if the path is a directory
-        if (!fDir && (dwEvent == SHCNE_DELETE) && m_dir_list.ContainsPath(szPath))
-        {
-            fDir = TRUE;
-            dwEvent = SHCNE_RMDIR;
```
Comment (learn-more) [inline dll/win32/shell32/shelldesktop/CDirectoryWatcher.cpp:179]: Is this event still created?
The shell expects this..

File: dll/win32/shell32/shelldesktop/CDirectoryWatcher.cpp [hunk 2]
```diff
@@ -171,37 +170,19 @@ void CDirectoryWatcher::ProcessNotification()
...
-        }
-
-        // update m_dir_list
         switch (dwEvent)
         {
             case SHCNE_MKDIR:
-                if (!PathIsDirectoryW(szPath) || !m_dir_list.AddPath(szPath))
```
Comment (learn-more) [inline dll/win32/shell32/shelldesktop/CDirectoryWatcher.cpp:186]: Previously this collapsed duplicate events, how is that handled now?

**Review by learn-more — CHANGES_REQUESTED** [review]
> Class `CDirectoryList` is unnecessary and slows down the system. JIRA issue: [CORE-13950](https://jira.reactos.org/browse/CORE-13950)

Did you profile it?
How did you come to the conclusion it is too slow?

**Comment by katahiromz** [comment]
> How did you come to the conclusion it is too slow?

Especially in the "recursive" case, the change notification is heavy.

File: dll/win32/shell32/shelldesktop/CDirectoryWatcher.cpp (hunk 1 above)
Comment (learn-more) [reply dll/win32/shell32/shelldesktop/CDirectoryWatcher.cpp:179]: https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&sr=1&s=SHCNE_RMDIR

There are multiple users of `SHCNE_RMDIR`

**Pull request closed without merging**